master/internal/rm/agentrm/provisioner/gcp_operation_tracker.go

```diff
@@ -26,171 +24,159 @@ type (
+			err := errors.Errorf(
 				"GCE cannot finish %q operation %q targeting %q",
 				resp.OperationType,
 				strconv.FormatUint(t.op.Id, 10),
 				t.op.TargetLink,
-			),
-			doneOp: resp,
-		}, nil
-	case resp.Status == "DONE":
-		// Stop tracking a operation when it's done and has no errors.
-		return &trackOperationDone{
-			op:     t.op,
-			doneOp: resp,
-		}, nil
-	case resp.Status == "RUNNING" || resp.Status == "PENDING":
-		return nil, nil
-	default:
-		errOp, _ := json.Marshal(resp)
-		return nil, errors.Errorf("unexpected message: %s", errOp)
+			)
+			done(t.op, resp, err)
+			return nil
+		case resp.Status == "DONE":
+			// Stop tracking a operation when it's done and has no errors.
+			done(t.op, resp, nil)
+			return nil
+		case resp.Status == "RUNNING" || resp.Status == "PENDING":
+			// Do nothing, keep tracking the operation.
+		default:
+			errOp, _ := json.Marshal(resp)
+			return errors.Errorf("unexpected message: %s", errOp)
+		}
+		// Slow down the polling rate.
+		time.Sleep(time.Second)
 	}
 }
 
 type gcpBatchOperationTracker struct {
+	mu sync.Mutex
+
 	config *provconfig.GCPClusterConfig
 	client *compute.Service
 
-	ops         []*compute.Operation
-	postProcess func(doneOps []*compute.Operation)
-	doneOps     []trackOperationDone
+	ops     []*compute.Operation
+	doneOps []trackOperationDone
+
+	syslog *logrus.Entry
 }
 
-func (t *gcpBatchOperationTracker) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		//nolint:lll // There isn't a great way to break this line that makes it more readable.
-		batchOperationTimeoutPeriod := time.Duration(len(t.ops)) * time.Duration(t.config.OperationTimeoutPeriod)
-		actors.NotifyAfter(ctx, batchOperationTimeoutPeriod, trackerTimeout{})
-		t.doneOps = make([]trackOperationDone, 0, len(t.ops))
-		for _, op := range t.ops {
-			if _, ok := ctx.ActorOf(
-				fmt.Sprintf("track-operation-%d", op.Id),
-				&gcpOperationTracker{
-					config:  t.config,
-					client:  t.client,
-					op:      op,
-					timeout: time.Duration(t.config.OperationTimeoutPeriod),
-				},
-			); !ok {
-				return errors.New("internal error tracking GCP operation")
-			}
-		}
+func newGCPBatchOperationTracker(
+	config *provconfig.GCPClusterConfig,
+	client *compute.Service,
+	ops []*compute.Operation,
+) *gcpBatchOperationTracker {
+	return &gcpBatchOperationTracker{
+		config:  config,
+		client:  client,
+		ops:     ops,
+		doneOps: make([]trackOperationDone, 0, len(ops)),
+		syslog:  logrus.WithField("component", "gcp-batch-operation-tracker"),
+	}
+}
 
-	case trackOperationDone:
-		if msg.err != nil {
-			ctx.Log().WithError(msg.err).Error("")
+func (t *gcpBatchOperationTracker) startTracker(postProcess func([]*compute.Operation)) {
+	timeout := time.Duration(t.config.OperationTimeoutPeriod)
+	groupCtx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	g := new(errgroup.Group)
```

Comment:
i think `var g errgroup.Group` is a way i see to use a zero value more frequently.